src/Adapter/MSTest.TestAdapter/Execution/TypeCache.cs

```diff
@@ -168,13 +170,8 @@ public IEnumerable<TestAssemblyInfo> AssemblyInfoListWithExecutableCleanupMethod
             }
 
             // Get the classInfo
-            classInfo = CreateClassInfo(type, testMethod);
-
-            // Use the full type name for the cache.
-            classInfo = _classInfoCache.GetOrAdd(typeName, classInfo);
-        }
-
-        return classInfo;
+            return @this.CreateClassInfo(type);
+        }, (this, testMethod));
```

Comment:
are we passing in this to avoid capturing it, or it has deeper meaning?

Reply:
Only to avoid capturing. That's why I explicitly added static modifier to the lambda.